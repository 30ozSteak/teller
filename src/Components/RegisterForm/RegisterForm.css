.register-box {
  height: 100vh;
  animation: registerAppear 0.5s ease-in;
}

@keyframes registerAppear {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

.register-header {
  width: 20rem;
  margin: 0 auto;
}

.register-header h1 {
  text-align: center;
  color: var(--TellerPink);
  font-family: teller;
  margin-top: 1rem;
}

.bottom-register-box {
  display: block;
  color: white;
  text-align: center;
  font-size: 0.8rem;
  position: absolute;
  bottom: 3rem;
}

.register-form {
  text-align: center;
  display: block;
  margin: 7rem auto 0;
}

.pls-password {
  color: rgba(255, 255, 255, 0.55);
  font-size: 0.5rem;
  margin-top: -1rem;
}

.register-form input {
  display: block;
  width: 15rem;
  height: 4rem;
  font-size: 1.5rem;
  border-left: none;
  border-top: none;
  border-right: none;
  border-bottom: 1px solid var(--TellerPink);
  background-color: transparent;
  margin: 1rem auto;
  color: white;
  text-align: center;
}

.register-form input::placeholder {
  text-align: center;
  color: white;
}

.register-confirm-input {
  border-top: 1px solid var(--TellerPink);
}

.register-button {
  color: white;
  background-color: transparent;
  border: none;
  border-radius: 10px;
  padding: 0.5rem 2rem;
  font-size: 1.5rem;
  background-color: var(--TellerPink);
}

.go-back-button {
  color: white;
  text-decoration: none;
  margin-top: 3rem;
  display: block;
}
