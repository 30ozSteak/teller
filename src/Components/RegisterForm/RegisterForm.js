import React, { Component } from "react";
import { registerUser } from "../../Utils/API";
import "./RegisterForm.css";
import { NavLink, Redirect } from "react-router-dom";

class RegisterForm extends Component {
  constructor(props) {
    super(props);

    const { storeToken } = this.props

    this.state = {
      name: "",
      email: "",
      password: "",
      confirmedPassword: "",
      passwordError: false,
      token: ""
    };
  }

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  handleSubmit = async e => {
    e.preventDefault();
    const { email, password, confirmedPassword } = this.state;
    const user = JSON.stringify({
      user: {
        email: email,
        password: password,
        password_confirmation: confirmedPassword
      }
    });
    const token = await registerUser(user);
    this.checkPassword();
    this.props.storeToken(token);
    this.setState({
      name: "",
      email: "",
      password: "",
      confirmedPassword: "",
      token: token.teller_api_token
    });
  };

  checkPassword = () => {
    const { password, confirmedPassword } = this.state;
    if (!password || !confirmedPassword || password !== confirmedPassword) {
      this.togglePasswordError();
    }
  };

  togglePasswordError = () => {
    const { passwordError } = this.state;
    this.setState({ passwordError: !passwordError });
  };

  render() {
    const { name, email, password, confirmedPassword, token } = this.state;

    if (token !== "" && typeof token !== "object") {
      return <Redirect to="/home" />
    }
    return (
      <div className="register-box">
        <div className="register-header">
          <h1> Register </h1>
        </div>
        <form className="register-form" onSubmit={this.handleSubmit}>
          <input
            className="register-input register-name-input"
            placeholder="Enter Full Name"
            type="text"
            name="name"
            onChange={this.handleChange}
            value={name}
          />
          <input
            className="register-input register-email-input"
            placeholder="Enter Email Address"
            type="email"
            name="email"
            onChange={this.handleChange}
            value={email}
          />
          <input
            className="register-input register-password-input"
            placeholder="Enter Password"
            type="password"
            name="password"
            onChange={this.handleChange}
            value={password}
          />
          <p className="pls-password">(At least 10 characters please)</p>
          <input
            className="register-input register-confirm-input"
            placeholder="Confirm Password"
            type="password"
            name="confirmedPassword"
            onChange={this.handleChange}
            value={confirmedPassword}
          />
          <button className="register-button">Register</button>
          <NavLink to="/" className="go-back-button">
            <p>Go Back</p>
          </NavLink>
        </form>
        <div className="bottom-register-box">
          <p>
            (An account is not necessary to use the majority of functionality
            our app provides.)
          </p>
        </div>
      </div>
    );
  }
}

export default RegisterForm;
