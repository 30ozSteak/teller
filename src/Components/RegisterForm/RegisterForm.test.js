import React from 'react'
import RegisterForm from './RegisterForm'
import { shallow, mount } from 'enzyme'

describe('RegisterForm Component', () => {
  let wrapper
  const mockUser = {
    name: 'Steve Jobs',
    id: 1234,
    email: 'steve.jobs@apple.com',
    password: 'apple'
  }

  beforeEach(() => {
    wrapper = shallow(<RegisterForm />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should start with the correct default state', () => {
    expect(wrapper.state().name).toEqual('')
    expect(wrapper.state().email).toEqual('')
    expect(wrapper.state().password).toEqual('')
    expect(wrapper.state().confirmedPassword).toEqual('')
    expect(wrapper.state().passwordError).toEqual(false)
  })

  describe('handleChange function', () => {
    it('should set state with the correct name', () => {
      const nameInput = wrapper.find('.register-name-input')
      nameInput.simulate('change', {
        target: { name: 'name', value: 'Steve Jobs' }
      })
      expect(wrapper.state().name).toEqual('Steve Jobs')
    })

    it('should set state with the correct email', () => {
      const emailInput = wrapper.find('.register-email-input')
      emailInput.simulate('change', {
        target: { name: 'email', value: 'steve.jobs@apple.com' }
      })
      expect(wrapper.state().email).toEqual('steve.jobs@apple.com')
    })

    it('should set state with the correct password', () => {
      const passwordInput = wrapper.find('.register-password-input')
      passwordInput.simulate('change', {
        target: { name: 'password', value: 'mypassword' }
      })
      expect(wrapper.state().password).toEqual('mypassword')
    })

    it('should set state with the correct confirmed password', () => {
      const confirmedInput = wrapper.find('.register-confirm-input')
      confirmedInput.simulate('change', {
        target: { name: 'confirmedPassword', value: 'mypassword' }
      })
      expect(wrapper.state().confirmedPassword).toEqual('mypassword')
    })

    it('should toggle error display if there is no password', () => {
      const passwordInput = wrapper.find('.register-password-input')
      const submitBtn = wrapper.find('.register-button')
      passwordInput.value = ''
      submitBtn.simulate('click')
      expect(wrapper.state().passwordError).toBe(true)
    })

    it('should toggle error display if there is no confirmed password', () => {

    })

    it('should toggle error display if the confirmedPassword and password do not match', () => {
      const passwordInput = wrapper.find('.register-password-input')
      const confirmedInput = wrapper.find('.register-confirm-input')
      const submitBtn = wrapper.find('.register-button')

      passwordInput.value = 'mypassword'
      confirmedInput.value = 'yourpassword'
      submitBtn.simulate('click')
      console.log(wrapper.state())
      expect(wrapper.state().passwordError).toEqual(true)
    })
  })

  describe('handleSubmit function', () => {
    let mockForm

    beforeEach(() => {
      mockForm = wrapper.find('.register-form')
    })

    xit('should submit user info to database', () => {

    })

    xit('should display an error if user information is incomplete', () => {

    })
  })
})