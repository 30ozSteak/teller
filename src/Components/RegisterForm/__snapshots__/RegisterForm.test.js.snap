// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RegisterForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="register-input register-name-input"
          name="name"
          onChange={[Function]}
          placeholder="Enter Full Name"
          type="text"
          value=""
        />,
        <input
          className="register-input register-email-input"
          name="email"
          onChange={[Function]}
          placeholder="Enter Email Address"
          type="email"
          value=""
        />,
        <input
          className="register-input register-password-input"
          name="password"
          onChange={[Function]}
          placeholder="Enter Password"
          type="password"
          value=""
        />,
        <input
          className="register-input register-confirm-input"
          name="confirmedPassword"
          onChange={[Function]}
          placeholder="Confirm Password"
          type="password"
          value=""
        />,
        <button
          className="register-button"
        >
          Register
        </button>,
      ],
      "className": "register-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "register-input register-name-input",
          "name": "name",
          "onChange": [Function],
          "placeholder": "Enter Full Name",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "register-input register-email-input",
          "name": "email",
          "onChange": [Function],
          "placeholder": "Enter Email Address",
          "type": "email",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "register-input register-password-input",
          "name": "password",
          "onChange": [Function],
          "placeholder": "Enter Password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "register-input register-confirm-input",
          "name": "confirmedPassword",
          "onChange": [Function],
          "placeholder": "Confirm Password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Register",
          "className": "register-button",
        },
        "ref": null,
        "rendered": "Register",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="register-input register-name-input"
            name="name"
            onChange={[Function]}
            placeholder="Enter Full Name"
            type="text"
            value=""
          />,
          <input
            className="register-input register-email-input"
            name="email"
            onChange={[Function]}
            placeholder="Enter Email Address"
            type="email"
            value=""
          />,
          <input
            className="register-input register-password-input"
            name="password"
            onChange={[Function]}
            placeholder="Enter Password"
            type="password"
            value=""
          />,
          <input
            className="register-input register-confirm-input"
            name="confirmedPassword"
            onChange={[Function]}
            placeholder="Confirm Password"
            type="password"
            value=""
          />,
          <button
            className="register-button"
          >
            Register
          </button>,
        ],
        "className": "register-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "register-input register-name-input",
            "name": "name",
            "onChange": [Function],
            "placeholder": "Enter Full Name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "register-input register-email-input",
            "name": "email",
            "onChange": [Function],
            "placeholder": "Enter Email Address",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "register-input register-password-input",
            "name": "password",
            "onChange": [Function],
            "placeholder": "Enter Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "register-input register-confirm-input",
            "name": "confirmedPassword",
            "onChange": [Function],
            "placeholder": "Confirm Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Register",
            "className": "register-button",
          },
          "ref": null,
          "rendered": "Register",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
